internal/lightcone/preservation/textureofmemories/textureofmemories.go

```diff
@@ -87,6 +87,7 @@ func onAfterBeingAttacked(mod *modifier.ModifierInstance, e event.AttackEndEvent
 		mod.Engine().AddModifier(mod.Owner(), info.Modifier{
 			Name:     modshieldbuff,
 			Source:   mod.Owner(),
+			State:    mod.State().(State),
```

Comment:
would passing state down like this work ? For `modshieldbuff`'s `onAdd` to get the required state, i give the adder function the current state in `mod`'s `onAfterBeingAttacked` like passing props in js